packages/browser-wallet/src/popup/pages/Account/Earn/Baking/Baking.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useContext, useEffect, useMemo } from 'react';
+import { Link, Route, Routes, useNavigate } from 'react-router-dom';
+import {
+    AccountInfoBakerV1,
+    AccountTransactionType,
+    isBakerAccountV1,
+    StakePendingChange,
+    StakePendingChangeType,
+    StakePendingChangeV1,
+} from '@concordium/web-sdk';
+import {
+    dateFromStakePendingChange,
+    displayAsCcd,
+    getFormattedDateString,
+    useIsSubsequentRender,
+} from 'wallet-common-helpers';
+import { useTranslation } from 'react-i18next';
+import { useAtomValue } from 'jotai';
+
+import { useSelectedAccountInfo } from '@popup/shared/AccountInfoListenerContext/AccountInfoListenerContext';
+import { ensureDefined } from '@shared/utils/basic-helpers';
+import Button from '@popup/shared/Button';
+import CheckmarkIcon from '@assets/svg/logo-checkmark.svg';
+import { selectedPendingTransactionsAtom } from '@popup/store/transactions';
+import LoadingIcon from '@assets/svg/pending-arrows.svg';
+import { BrowserWalletAccountTransaction, TransactionStatus } from '@popup/shared/utils/transaction-history-types';
+import ButtonGroup from '@popup/shared/ButtonGroup';
+import { openStatusToDisplay } from '@popup/shared/utils/baking-helpers';
+import DisplayPartialString from '@popup/shared/DisplayPartialString';
+import RegisterBaking from './RegisterBaking';
+import RemoveBaking from './RemoveBaking';
+import UpdateBaking from './UpdateBaking';
+import { earnPageContext } from '../utils';
+
+const routes = {
+    register: 'register',
+    update: 'update',
+    remove: 'remove',
+};
+
+type PendingBakingProps = {
+    transaction: BrowserWalletAccountTransaction;
+};
+
+function PendingBaking({ transaction }: PendingBakingProps): JSX.Element {
+    const { t } = useTranslation('account', { keyPrefix: 'baking.details' });
+
+    return (
+        <div className="flex-column align-center justify-center h-full">
+            {transaction.status !== TransactionStatus.Failed && <LoadingIcon className="loading" />}
+            <h3 className="m-t-5 m-b-0 m-h-30 text-center">
+                {t(transaction.status === TransactionStatus.Failed ? 'failed' : 'pending')}
+            </h3>
+        </div>
+    );
+}
+
+type DisplayPendingChangeProps = {
+    pendingChange?: StakePendingChange;
+};
+
+function DisplayPendingChange({ pendingChange }: DisplayPendingChangeProps) {
+    const { t: sharedT } = useTranslation('shared', { keyPrefix: 'baking' });
+    const { consensusStatus, tokenomicsInfo, chainParameters } = useContext(earnPageContext);
+
+    const effectiveTime = useMemo(() => {
+        if (pendingChange) {
+            const date = dateFromStakePendingChange(pendingChange, consensusStatus, tokenomicsInfo, chainParameters);
+            if (date) {
+                return getFormattedDateString(date);
+            }
+        }
+        return undefined;
+    }, [
+        (pendingChange as StakePendingChangeV1)?.effectiveTime?.toString?.(),
```

Comment:
Could this be a separate variable instead? Seems weird to me to have a function call be part of the dependencies list